test/interoperability/interop_test_suite.py

```diff
@@ -0,0 +1,537 @@
+                    returned_subs.add(sub["subscription_id"])
+
+        missing_subs = all_sub_2 - returned_subs
+        assert (
+            missing_subs == set()
+        ), f"{dss} returned too few Subscriptions, missing: {missing_subs}"
+
+        extra_subs = returned_subs - all_sub_2
+        assert (
+            extra_subs == set()
+        ), f"{dss} returned too many Subscriptsions, extra: {extra_subs}"
+
+    def testStep11(
+        self, dss_map: Dict[str, clients.DSSClient], primary_dss: str, **kwargs
+    ) -> None:
+        """ISA deletion triggers subscription notification requests"""
+        resp = dss_map[primary_dss].delete(
+            f"/identification_service_areas/{self.context['isa_2_uuid'].key}/"
+        )
+        assert (
+            resp.status_code == 200
+        ), f"Failed to delete ISA to {primary_dss}. Error: {resp.json()['error']}"
+
+        all_sub_2 = set()
+        for name, context in self.context.items():
+            if name.startswith("sub_2_"):
+                all_sub_2.add(context.key)
+
+        returned_subs = set()
+        for subscriber in resp.json()["subscribers"]:
+            for subscription in subscriber:
+                for sub in subscriber["subscriptions"]:
+                    returned_subs.add(sub["subscription_id"])
+
+        missing_subs = all_sub_2 - returned_subs
+        assert (
+            missing_subs == set()
+        ), f"{dss} returned too few Subscriptions, missing: {missing_subs}"
+
+        extra_subs = returned_subs - all_sub_2
+        assert (
+            extra_subs == set()
+        ), f"{dss} returned too many Subscriptsions, extra: {extra_subs}"
+
+    def testStep12(
+        self, dss_map: Dict[str, clients.DSSClient], primary_dss: str, **kwargs
+    ) -> None:
+        """Expired Subscriptions don’t trigger subscription notification requests"""
+        time.sleep(SHORT_WAIT_SEC)
+        time_start = datetime.datetime.utcnow()
+        time_end = time_start + datetime.timedelta(minutes=10)
+        self.context["isa_3_uuid"] = TestContext("ISA", str(uuid.uuid4()))
+        resp = dss_map[primary_dss].put(
+            f"/identification_service_areas/{self.context['isa_3_uuid'].key}",
+            json={
+                "extents": {
+                    "spatial_volume": {
+                        "footprint": {"vertices": VERTICES},
+                        "altitude_lo": 20,
+                        "altitude_hi": 400,
+                    },
+                    "time_start": time_start.strftime(DATE_FORMAT),
+                    "time_end": time_end.strftime(DATE_FORMAT),
+                },
+                "flights_url": "https://example.com/dss",
+            },
+        )
+        assert (
+            resp.status_code == 200
+        ), f"Failed to insert ISA to {primary_dss}. Error: {resp.json()['error']}"
+        subscribers = resp.json()["subscribers"]
+
+        assert (
+            subscribers == []
```

Comment:
To be as robust as possible, we actually only care that none of the known-expired Subscriptions are included. There could perhaps be pre-existing Subscriptions.